### DISCUSSION TITLE
Invalid HTTP header: Content-Type=undefined when using zValidator with JSON schema

### DISCUSSION THREAD

**Discussion by ilshm** [colinhacks/zod#3478] [Q&A]
I am encountering an issue when using `zValidator` with a JSON schema in my Node.js application. Specifically, I am receiving the error "Invalid HTTP header: Content-Type=undefined" when trying to validate JSON data using the following schema and validation setup:

```javascript
const schema = z.object({
  accessCode: z.string().optional()
})

zValidator(
  'json',
  schema
)
```

What I want to achieve:
1. Validate if the provided `accessCode` is a boolean.
2. Allow the request to proceed even if the request body is not provided.
3. Suppress all Zod errors and return only a 400 status code without any body in case of validation failure.

Is it possible to achieve these requirements with Zod and zValidator? If so, how can I modify my setup to accomplish this? If not, are there any workarounds or alternative approaches I can consider?

Thank you.

**Comment by stefnba** [comment]
Running into the same issue.

**Comment by movahhedi** [comment]
Me too. I'm using [Hono](https://hono.dev/guides/validation#zod-validator-middleware) as webserver.

**Comment by 5ika** [comment]
Same with Hono. To avoid error, provide Content-Type header + some JSON payload as body.

Example:

```bash
 curl --location 'http://localhost:8000/notes' \
--header 'Content-Type: application/json' \
--data '{"accessCode": "12"}'
```
